com.unity.perception/Runtime/GroundTruth/Labeling/LabelManager.cs

```diff
@@ -0,0 +1,168 @@
+
+                RecursivelyInitializeGameObjects(
+                    unregisteredLabel.gameObject,
+                    new MaterialPropertyBlock(),
+                    unregisteredLabel,
+                    instanceId);
+
+                unregisteredLabel.SetInstanceId(instanceId);
+                m_RegisteredLabels.Add(unregisteredLabel);
+            }
+
+            m_LabelsPendingRegistration.Clear();
+        }
+
+        /// <summary>
+        /// Activates the given <see cref="IGroundTruthGenerator"/>.
+        /// <see cref="IGroundTruthGenerator.SetupMaterialProperties"/> will be called for all
+        /// <see cref="MeshRenderer"/> instances under each object containing a <see cref="Labeling"/> component.
+        /// </summary>
+        /// <param name="generator">The generator to register</param>
+        public void Activate(IGroundTruthGenerator generator)
+        {
+            m_ActiveGenerators.Add(generator);
+            foreach (var label in m_RegisteredLabels)
+                RecursivelyInitializeGameObjects(label.gameObject, new MaterialPropertyBlock(), label, label.instanceId);
+        }
+
+        /// <summary>
+        /// Deactivates the given <see cref="IGroundTruthGenerator"/>.
+        /// It will no longer receive calls when <see cref="Labeling"/> instances are created.
+        /// </summary>
+        /// <param name="generator">The generator to deactivate</param>
+        /// <returns>
+        /// True if the <see cref="generator"/> was successfully removed.
+        /// False if <see cref="generator"/> was not active.
+        /// </returns>
+        public bool Deactivate(IGroundTruthGenerator generator)
+        {
+            return m_ActiveGenerators.Remove(generator);
+        }
+
+        /// <summary>
+        /// Registers a labeling component
+        /// </summary>
+        /// <param name="labeledObject">the component to register</param>
+        internal void Register(Labeling labeledObject)
+        {
+            m_LabelsPendingRegistration.Add(labeledObject);
+        }
+
+        /// <summary>
+        /// Unregisters a labeling component
+        /// </summary>
+        /// <param name="labeledObject">the component to unregister</param>
+        internal void Unregister(Labeling labeledObject)
+        {
+            m_LabelsPendingRegistration.Remove(labeledObject);
+            m_RegisteredLabels.Remove(labeledObject);
+        }
+
+        /// <summary>
+        /// Refresh ground truth generation for the labeling of a particular GameObject. This is necessary when the
+        /// list of labels changes or when renderers or materials change on objects in the hierarchy.
+        /// </summary>
+        /// <param name="labeledObject">the component to refresh</param>
+        internal void RefreshLabeling(Labeling labeledObject)
+        {
+            m_RegisteredLabels.Remove(labeledObject);
+            m_LabelsPendingRegistration.Add(labeledObject);
+        }
```

Comment:
perhaps rename `labeledObject` to `labeling` to better reflect the type of the function arguments?

Reply:
done